### PR TITLE
Resend confirmation

### LINKED ISSUE #237: Send email confirmation again
Hello it is me again :)

I think it would be useful to add in `/auth/activate` options to generate a new ticket when the old one expires and send a confirmation message again, or to automatically delete an unconfirmed account and give the opportunity to create a new one.

Example: Registers in the application, but I can't log in to the email inbox right now. Once the ticket's validity has expired, you cannot activate your account, you cannot log in to your account, or you cannot create a new account at this email address.

Greetings.

### PR DESCRIPTION
**Pull request by komninoschat**
Resolves #237.

### CODE REVIEW

File: db/migrations/00002_add_confirmation_reset_timeout.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE auth.accounts
+  ADD COLUMN last_sent_confirmation timestamp with time zone DEFAULT now() NOT NULL;
```
Comment (elitan) [inline db/migrations/00002_add_confirmation_reset_timeout.sql:2]: we use `<verb>_at` ex `updated_at` or `created_at` - so I'm suggesting we use `confirmation_email_sent_at` or `last_confirmation_email_sent_at`.

File: src/routes/auth/resend-confirmation.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const account = await selectAccount(body)
+
+  if (!account) {
+    return res.boom.badRequest('Account does not exist.')
+  } else if (account.active) {
+    return res.boom.badRequest('Account already activated.')
+  } else if (
+    +new Date(
+      +new Date(account.last_sent_confirmation) + REGISTRATION.CONFIRMATION_RESET_TIMEOUT
+    ) > +new Date()
```
Comment (elitan) [inline src/routes/auth/resend-confirmation.ts:25]: Can we make this code more readable?

**Review by elitan — CHANGES_REQUESTED** [review]

**Review by elitan — CHANGES_REQUESTED** [review]
There is a lot of discussions in https://github.com/nhost/hasura-backend-plus/issues/237.

One of the suggestions was to adopt this into `/auth/register` and let that endpoint send out a new email (with a timeout) if there is already a user that is inactive.

If we keep this `/auth/resend-confirmation` endpoint, there is no way of telling of a user was already registered or not since we just return "Account already exists" if you try to register an account with an email that's already registered (https://github.com/nhost/hasura-backend-plus/blob/master/src/routes/auth/register.ts#L26-L28).

Do we have any references on how other services and/or auth services handles this?

**Comment by joshmedeski** [comment]
@elitan any ETA on when this feature will be published?

**Pull request merged**